### PR TITLE
Delete project

### LINKED ISSUE #66: Delete project
Delete project, related entries and images

### PR DESCRIPTION
**Pull request by zli117**
Added a delete project endpoint. The endpoint is `DELETE /project/<project_id>`
Closes #66

### CODE REVIEW

File: instadam/project.py [hunk 1]
```diff
@@ -319,3 +324,39 @@ def update_user_permission(project_id):
         db.session.commit()
 
     return construct_msg('Permission added successfully'), 200
+
+
+@bp.route('/project/<project_id>', methods=['DELETE'])
+@jwt_required
+def delete_user(project_id):
```
Comment (enlne) [inline instadam/project.py]: this function does not have a comment

File: tests/test_project.py [hunk 2]
```diff
@@ -64,17 +74,18 @@ def test_project_repr(client):
...
+        headers={'Authorization': 'Bearer %s' % token})
     assert response.status_code == 201
     res = response.get_json()
     assert 'project_id' in res
 
 
 def test_create_project_not_logged_in(client):
-    response = client.post(PROJECT_ENDPOINT,
-                           json={'project_name': FAIL_PROJECT_NAME})
+    response = client.post(
```
Comment (enlne) [inline tests/test_project.py:90]: previously, you had one parameter per line, but here you have two parameters in the same line

File: tests/test_project.py [hunk 3]
```diff
@@ -136,13 +151,41 @@ def get_project_fixture():
...
+        label1 = Label(label_name='asdf1', project_id=project.id)
+        db.session.add(label1)
+        label2 = Label(label_name='asdf2', project_id=project.id)
+        db.session.add(label2)
+        db.session.flush()
+        db.session.commit()
+
+        annotation1 = Annotation(
+            image_id=1, project_id=1, creator_id=1, label_id=1, data=b'1234',
+            vector=b'1234')
```
Comment (enlne) [inline tests/test_project.py:175]: why does this vector get its own line?

File: instadam/project.py [hunk 4]
```diff
@@ -319,3 +324,39 @@ def update_user_permission(project_id):
...
+    project = maybe_get_project(project_id)  # check privilege and get project
+
+    # Delete images
```
Comment (enlne) [inline instadam/project.py]: Your other comments are good, but I don't think you need these delete comments because the code is short and clear

**Review by enlne — COMMENTED** [review]
Our checklist:
- variable names that are not generic
- function names that explain what they do 
- appropriate comments 
- consistency when listing parameters
- cohesion on which functions should be in a file 
- method length

File: tests/test_project.py [hunk 5]
```diff
@@ -1,6 +1,11 @@
+import os
```
Comment (enlne) [inline tests/test_project.py:4]: Some comments in the testing file would be helpful

File: tests/test_project.py (hunk 2 above)
Comment (nishanthrs) [reply tests/test_project.py:90]: The coding style as dictated by the Google Python Style Guide (http://google.github.io/styleguide/pyguide.html) says that there must not be more than 80 chars per line. However, it's true that it's not consistent with the other lines in the file, so we will change it.

File: tests/test_project.py (hunk 3 above)
Comment (nishanthrs) [reply tests/test_project.py:175]: The coding style as dictated by the Google Python Style Guide (http://google.github.io/styleguide/pyguide.html) says that there must not be more than 80 chars per line. That's why the vector property is moved to a new line.

File: instadam/project.py (hunk 1 above)
Comment (PrahladSrikant) [reply instadam/project.py]: We will add comments to document this function

File: instadam/project.py (hunk 4 above)
Comment (zli117) [reply instadam/project.py]: Acknowledged

File: tests/test_project.py (hunk 5 above)
Comment (PrahladSrikant) [reply tests/test_project.py:4]: We will add some comments in this testing file

**Review by enlne — COMMENTED** [review]
line 125, 167
Just noting that the function descriptions are not consistent

**Comment by zli117** [comment]
> line 125, 167
> Just noting that the function descriptions are not consistent

Could you update the line number since we merged with master? Thanks!

**Pull request merged**